### ISSUE TITLE
Bumped RedisGraph from 2.8.4 to 2.8.19 and now getting 'RedisGraph used forced unlocking commit flow for the query' warning

### ISSUE THREAD

**Issue by blockchain-art** [RedisGraph/RedisGraph#2587]
We've bumped our RedisGraph docker image from `2.8.4` to `2.8.19` and now are getting a bunch of these warning messages:

```
RedisGraph used forced unlocking commit flow for the query
MERGE (m:Measure {value:238504})
MERGE (measurement:Measurement {storage_key:"mi238504|cdsc1ab5a92|s00192ee2"})
MERGE (surveydatacollectorid197475:Scope {storage_key:"scope|sad5f9bfc|sf1fdc72f|i197475",value:197475})
ON CREATE SET surveydatacollectorid197475 += {entity:"survey",attribute:"data_collector_id"}
MERGE (surveydatacollectorid197475)-[:SCOPED]->(measurement)
MERGE (surveyreportid345631:Scope {storage_key:"scope|sad5f9bfc|s4bd2a4c0|i345631",value:345631})
ON CREATE SET surveyreportid345631 += {entity:"survey",attribute:"report_id"}
MERGE (surveyreportid345631)-[:SCOPED]->(measurement)
MERGE (surveycountrycodeZA:Scope {storage_key:"scope|sad5f9bfc|sf026bb7c|s0056d627",value:"ZA"})
ON CREATE SET surveycountrycodeZA += {entity:"survey",attribute:"country_code"}
MERGE (surveycountrycodeZA)-[:SCOPED]->(measurement)
MERGE (surveycountrySouthAfrica:Scope {storage_key:"scope|sad5f9bfc|s5373c966|sa2a3d8cc",value:"South Africa"})
ON CREATE SET surveycountrySouthAfrica += {entity:"survey",attribute:"country"}
MERGE (surveycountrySouthAfrica)-[:SCOPED]->(measurement)
MERGE (surveyorderid437618:Scope {storage_key:"scope|sad5f9bfc|s8d9f6d38|i437618",value:437618})
ON CREATE SET surveyorderid437618 += {entity:"survey",attribute:"order_id"}
MERGE (surveyorderid437618)-[:SCOPED]->(measurement)
MERGE (surveypanelsupplierid4:Scope {storage_key:"scope|sad5f9bfc|s06bde5b2|i4",value:4})
ON CREATE SET surveypanelsupplierid4 += {entity:"survey",attribute:"supplier_id"}
MERGE (surveypanelsupplierid4)-[:SCOPED]->(measurement)
MERGE (productid18993:Scope {storage_key:"scope|sd34a04ad|sbf396750|i18993",value:18993})
ON CREATE SET productid18993 += {entity:"product",attribute:"id"}
MERGE (productid18993)-[:SCOPED]->(measurement)
MERGE (productcodecontentstoryboard:Scope {storage_key:"scope|sd34a04ad|s77153098|s0ae78180",value:"content_storyboard"})
ON CREATE SET productcodecontentstoryboard += {entity:"product",attribute:"code"}
MERGE (productcodecontentstoryboard)-[:SCOPED]->(measurement)
MERGE (cl1yft:Constant {value:5149.0})
MERGE (measurement)-[:MEASURES {dimension: 'choice'}]->(cl1yft)
MERGE (m)-[l:CAPTURES {storage_key: 'response_id|mi238504|cdsc1ab5a92|s00192ee2'}]->(measurement)
RETURN distinct l.storage_key
```

note the `RedisGraph used forced unlocking commit flow for the query`

**Comment by blockchain-art** [comment]
happening with every query pretty much

**Comment by LiorKogan** [comment]
@blockchain-art I'm unable to reproduce it just by running this single query.

**Comment by blockchain-art** [comment]
I cannot reproduce it either by running single queries - we've got a job that essentially writes queries such as the above to the graph, and in the logs every query is prefixed with said warning message. Not sure what to do or what its implications are

**Comment by AviAvni** [comment]
are you sure you use RedisGraph 2.8.19?

**Comment by blockchain-art** [comment]
> are you sure you use RedisGraph 2.8.19?

Switched to using this docker image `redislabs/redisgraph:2.8.19`

**Comment by AviAvni** [comment]
and the warning produced by RedisGraph 2.8.19?

**Comment by blockchain-art** [comment]
> and the warning produced by RedisGraph 2.8.19?

Well from my perspective we bumped RedisGraph image from version `2.8.4` to `2.8.19` and the warning message started appearing with every query, so I would guess so? we've got no other instance of it prior when I look at Elasticsearch which is where write logs to

**Comment by blockchain-art** [comment]
I can replicate it on an M1 laptop too actually, when using: `FROM redisfab/redisgraph:2.8.19-arm64v8-bionic`, but if I upgrade to `FROM redisfab/redisgraph:master-arm64v8-bionic` then the warning disappears


Here's the full log of the sequence of queries that trigger it
```
1662975370.848094 [1 172.19.0.1:63378] "GRAPH.QUERY" "hms_test" "MERGE (m:Measure {value:\"another_measure\"})\nMERGE (measurement:Measurement {storage_key:\"msf77e868e|cdsca9bc19c|i2\"})\nMERGE (source:Source:survey {source_id: \"attributereportid123\"})\nMERGE (attributereportid123:Scope {storage_key:\"scope|sfa7aeffb|s4bd2a4c0|i123\",value:123})\nON CREATE SET attributereportid123 += {entity:\"attribute\",attribute:\"report_id\"}\nMERGE (source)-[:HAS]->(attributereportid123)\nMERGE (source)-[:SCOPED]->(measurement)\nMERGE (source)-[:CAPTURES]->(m)\nMERGE (cqsa59:Constant {value:2})\nMERGE (measurement)-[:MEASURES {dimension: 'dimension'}]->(cqsa59)\nMERGE (m)-[l:CAPTURES {storage_key: 'response_id|msf77e868e|cdsca9bc19c|i2'}]->(measurement)RETURN distinct l.storage_key" "--compact"
1662975370.851955 [1 172.19.0.1:63378] "GRAPH.QUERY" "hms_test" "MATCH (m:Measure {value:\"another_measure\"})-[:EQUALS]->(:Measure)-[:CAPTURES]->(n:Measurement), (me:Measurement {storage_key:\"msf77e868e|cdsca9bc19c|i2\"}),(n)-[d:MEASURES]->(c:Constant)\n      WITH n, collect(c) as c, collect(d) as d, me\n      WHERE all(const IN c WHERE const.value IN [2]) AND all(dim IN d WHERE dim.dimension IN [\"dimension\"])\n      AND NOT id(me) = id(n)\n      WITH distinct collect(n) as n, me\n      MATCH (n)-[:EQUALS]-(other_m:Measurement)\n      UNWIND n as n\n      MERGE (me)-[:EQUALS]->(other_m)\n      RETURN count(me)" "--compact"
1662975370.855048 [1 172.19.0.1:63378] "GRAPH.QUERY" "hms_test" "MERGE (m:Measure {value:\"test_measure\"})\nMERGE (measurement:Measurement {storage_key:\"ms5c46cded|cdsca9bc19c|i1\"})\nMERGE (source:Source:survey {source_id: \"attributereportid125\"})\nMERGE (attributereportid125:Scope {storage_key:\"scope|sfa7aeffb|s4bd2a4c0|i125\",value:125})\nON CREATE SET attributereportid125 += {entity:\"attribute\",attribute:\"report_id\"}\nMERGE (source)-[:HAS]->(attributereportid125)\nMERGE (source)-[:SCOPED]->(measurement)\nMERGE (source)-[:CAPTURES]->(m)\nMERGE (czbrse:Constant {value:1})\nMERGE (measurement)-[:MEASURES {dimension: 'dimension'}]->(czbrse)\nMERGE (m)-[l:CAPTURES {storage_key: 'response_id|ms5c46cded|cdsca9bc19c|i1'}]->(measurement)RETURN distinct l.storage_key" "--compact"
1662975370.857590 [1 172.19.0.1:63378] "GRAPH.QUERY" "hms_test" "MATCH (m:Measure {value:\"test_measure\"})-[:EQUALS]->(:Measure)-[:CAPTURES]->(n:Measurement), (me:Measurement {storage_key:\"ms5c46cded|cdsca9bc19c|i1\"}),(n)-[d:MEASURES]->(c:Constant)\n      WITH n, collect(c) as c, collect(d) as d, me\n      WHERE all(const IN c WHERE const.value IN [1]) AND all(dim IN d WHERE dim.dimension IN [\"dimension\"])\n      AND NOT id(me) = id(n)\n      WITH distinct collect(n) as n, me\n      MATCH (n)-[:EQUALS]-(other_m:Measurement)\n      UNWIND n as n\n      MERGE (me)-[:EQUALS]->(other_m)\n      RETURN count(me)" "--compact"
1662975370.860402 [1 172.19.0.1:63378] "GRAPH.QUERY" "hms_test" "MERGE (m:Measure {value:\"test_measure\"})\nMERGE (measurement:Measurement {storage_key:\"ms5c46cded|cdsca9bc19c|i2\"})\nMERGE (source:Source:survey {source_id: \"attributereportid125\"})\nMERGE (attributereportid125:Scope {storage_key:\"scope|sfa7aeffb|s4bd2a4c0|i125\",value:125})\nON CREATE SET attributereportid125 += {entity:\"attribute\",attribute:\"report_id\"}\nMERGE (source)-[:HAS]->(attributereportid125)\nMERGE (source)-[:SCOPED]->(measurement)\nMERGE (source)-[:CAPTURES]->(m)\nMERGE (cdobs2:Constant {value:2})\nMERGE (measurement)-[:MEASURES {dimension: 'dimension'}]->(cdobs2)\nMERGE (m)-[l:CAPTURES {storage_key: 'response_id|ms5c46cded|cdsca9bc19c|i2'}]->(measurement)RETURN distinct l.storage_key" "--compact"
1662975370.863105 [1 172.19.0.1:63378] "GRAPH.QUERY" "hms_test" "MATCH (m:Measure {value:\"test_measure\"})-[:EQUALS]->(:Measure)-[:CAPTURES]->(n:Measurement), (me:Measurement {storage_key:\"ms5c46cded|cdsca9bc19c|i2\"}),(n)-[d:MEASURES]->(c:Constant)\n      WITH n, collect(c) as c, collect(d) as d, me\n      WHERE all(const IN c WHERE const.value IN [2]) AND all(dim IN d WHERE dim.dimension IN [\"dimension\"])\n      AND NOT id(me) = id(n)\n      WITH distinct collect(n) as n, me\n      MATCH (n)-[:EQUALS]-(other_m:Measurement)\n      UNWIND n as n\n      MERGE (me)-[:EQUALS]->(other_m)\n      RETURN count(me)" "--compact"
1662975370.865811 [1 172.19.0.1:63378] "GRAPH.QUERY" "hms_test" "MERGE (m:Measure {value:\"another_measure\"})\nMERGE (measurement:Measurement {storage_key:\"msf77e868e|cdsca9bc19c|i1\"})\nMERGE (source:Source:survey {source_id: \"attributereportid125\"})\nMERGE (attributereportid125:Scope {storage_key:\"scope|sfa7aeffb|s4bd2a4c0|i125\",value:125})\nON CREATE SET attributereportid125 += {entity:\"attribute\",attribute:\"report_id\"}\nMERGE (source)-[:HAS]->(attributereportid125)\nMERGE (source)-[:SCOPED]->(measurement)\nMERGE (source)-[:CAPTURES]->(m)\nMERGE (c23407:Constant {value:1})\nMERGE (measurement)-[:MEASURES {dimension: 'dimension'}]->(c23407)\nMERGE (m)-[l:CAPTURES {storage_key: 'response_id|msf77e868e|cdsca9bc19c|i1'}]->(measurement)RETURN distinct l.storage_key" "--compact"
1662975370.868489 [1 172.19.0.1:63378] "GRAPH.QUERY" "hms_test" "MATCH (m:Measure {value:\"another_measure\"})-[:EQUALS]->(:Measure)-[:CAPTURES]->(n:Measurement), (me:Measurement {storage_key:\"msf77e868e|cdsca9bc19c|i1\"}),(n)-[d:MEASURES]->(c:Constant)\n      WITH n, collect(c) as c, collect(d) as d, me\n      WHERE all(const IN c WHERE const.value IN [1]) AND all(dim IN d WHERE dim.dimension IN [\"dimension\"])\n      AND NOT id(me) = id(n)\n      WITH distinct collect(n) as n, me\n      MATCH (n)-[:EQUALS]-(other_m:Measurement)\n      UNWIND n as n\n      MERGE (me)-[:EQUALS]->(other_m)\n      RETURN count(me)" "--compact"
1662975370.871194 [1 172.19.0.1:63378] "GRAPH.QUERY" "hms_test" "MERGE (m:Measure {value:\"another_measure\"})\nMERGE (measurement:Measurement {storage_key:\"msf77e868e|cdsca9bc19c|i2\"})\nMERGE (source:Source:survey {source_id: \"attributereportid125\"})\nMERGE (attributereportid125:Scope {storage_key:\"scope|sfa7aeffb|s4bd2a4c0|i125\",value:125})\nON CREATE SET attributereportid125 += {entity:\"attribute\",attribute:\"report_id\"}\nMERGE (source)-[:HAS]->(attributereportid125)\nMERGE (source)-[:SCOPED]->(measurement)\nMERGE (source)-[:CAPTURES]->(m)\nMERGE (cfrr8m:Constant {value:2})\nMERGE (measurement)-[:MEASURES {dimension: 'dimension'}]->(cfrr8m)\nMERGE (m)-[l:CAPTURES {storage_key: 'response_id|msf77e868e|cdsca9bc19c|i2'}]->(measurement)RETURN distinct l.storage_key" "--compact"
1662975370.875209 [1 172.19.0.1:63378] "GRAPH.QUERY" "hms_test" "MATCH (m:Measure {value:\"another_measure\"})-[:EQUALS]->(:Measure)-[:CAPTURES]->(n:Measurement), (me:Measurement {storage_key:\"msf77e868e|cdsca9bc19c|i2\"}),(n)-[d:MEASURES]->(c:Constant)\n      WITH n, collect(c) as c, collect(d) as d, me\n      WHERE all(const IN c WHERE const.value IN [2]) AND all(dim IN d WHERE dim.dimension IN [\"dimension\"])\n      AND NOT id(me) = id(n)\n      WITH distinct collect(n) as n, me\n      MATCH (n)-[:EQUALS]-(other_m:Measurement)\n      UNWIND n as n\n      MERGE (me)-[:EQUALS]->(other_m)\n      RETURN count(me)" "--compact"
1662975370.878198 [1 172.19.0.1:63378] "GRAPH.QUERY" "hms_test" "\nMATCH (source:Source:survey)\nMERGE (respondentstatestateexportable:Scope {storage_key:\"scope|s3a8d3b8e|sa393d2fb|s3870bf82\",value:\"exportable\"})\nON CREATE SET respondentstatestateexportable += {entity:\"respondent_state\",attribute:\"state\"}\nMERGE (source)-[:HAS]->(respondentstatestateexportable)" "--compact"
```

**Comment by blockchain-art** [comment]
Have you been able to replicate this? should we upgrade to `master`? what are the side effects of this warning

**Comment by amanwswj** [comment]
This problem also occurs when I use version 2.8.19

**Comment by AviAvni** [comment]
we managed to fix this and will release 2.8.20 soon with this fix
